### DISCUSSION TITLE
Android build error: invalid source release: 17

### DISCUSSION THREAD

**Discussion by jacobdedman** [codemagic-ci-cd/codemagic-docs#2147] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [x] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Build for android
2. error

### Expected results

I expect the app to build

### Actual results

App build failed for android with error:

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':add_2_calendar:compileReleaseJavaWithJavac'.
> error: invalid source release: 17

This only started happening after I tried to do a build to support Android 14. But I get the same error when I try to build my main branch that has successfully built in the past. 

gradlew and graddle-wrapper files are in the repo I am building from

### Build id (optional)

65380254d594314bb3fb964c

**Comment by dominikIncibit** [comment]
same problem here! Is there any solution available?

**Reply by Ttkrommendijk** [reply]
same here. I use flutterflow and deploy to app store which is all fine, but when deploying directly to playstore I get the same message.

**Comment by jacobdedman** [comment]
@fleckdalm @Ttkrommendijk I think I found the solution:
For Android 14 you'll need these versions (or more recent ones)

Flutter: 3.7.12
Gradle: 8
Android Gradle Plugin: 7.3.0 - this is in android/build.gradle in the buildscript/dependencies block - "classpath 'com.android.tools.build:gradle:7.3.0'"
Kotlin Gradle Plugin: 1.6.10 - this is also in android/build.gradle - ext.kotlin_version = '1.6.10' - and the classpath -  "classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version""

I am still having local build issues with other dependencies from pubspec.yaml needing upgrades but I think the above is the general fix for this

**Comment by jacobdedman** [comment, marked as answer]
This was an issue with the add_2_calendar package

A recent update (2.2.5) the the add_2_calendar package added support for iOS 17 and I'm assuming the error log was in reference to that. Downgrading to add_2_calendar: 2.2.4 solved this issue

**Reply by Ttkrommendijk** [reply]
Check, will try it out here! Thanks for the reply!

**Reply by Ttkrommendijk** [reply]
I'm working with flutterflow. By any chance you would know how to downgrade here? 
I've changed the pubspec dependencies to ^2.2.4, I thought that would be sufficient.

Unfortunately still getting a built error for android( IOS is fine)

**Reply by jacobdedman** [reply]
You need to remove the ^. In pubspec a package defined as package-name: ^2.x.y allows any version >=2.x.y and <3.0.0. So defining it as ^2.2.4 still allows it to get any version >=2.2.4 and <3.0.0, so it grabs the highest version under 3.0.0 which is 2.2.5.

Fix:

add_2_calendar: 2.2.4

**Reply by jacobdedman** [reply]
You can check your pubspec.lock to see what versions of all your packages you are actually getting. If you check your add_2_calendar version in pubspec.lock with ^2.2.4 still in pubspec.yaml you will see that it is using 2.2.5. After changing it to 2.2.4 without ^ you should see it go down to 2.2.4 in pubspec.lock
